docs/content/en/highlights/2021-07-13-helm-customConfig.md

```diff
@@ -0,0 +1,319 @@
+
+## Upgrade Guide
+
+### globalOptions and logSchema
+
+Any [global options](https://vector.dev/docs/reference/configuration/global-options/) can be moved directly
+into the `customConfig` key and converted into snake-case.
+
+```diff title="values.yaml"
+   globalOptions:
++    enabled: false
+     dataDir: "/vector-data-dir"
+   logSchema:
++    enabled: false
+     hostKey: "host"
+     messageKey: "message"
+     sourceTypeKey: "source_type"
+     timestampKey: "timestamp"
+   ...
++  customConfig:
++    data_dir: "/vector-data-dir"
++    log_schema:
++      host_key: host
++      message_key: message
++      source_type_key: source_type
++      timestamp_key: timestamp
++  ...
+```
+
+### vectorApi
+
+[Vector API](https://vector.dev/docs/reference/api/) configuration can be moved as is under a `customConfig.api` key.
+The `extraContainerPorts` or `service` key should be used to expose the port configured in `customConfig`.
+
+```diff title="values.yaml"
+   vectorApi:
++    enabled: false
+     address: "0.0.0.0:8686"
+     playground: true
+   ...
++  customConfig:
++    ...
++    api:
++      enabled: true
++      address: 127.0.0.1:8686
++      playground: true
++    ...
+```
+
+### kubernetesLogsSource
+
+The vector-agent chart will continue to mount the hostPaths required to access Pod logs.
+
+```diff title="values.yaml"
+   kubernetesLogsSource:
++    enabled: false
+     sourceId: kubernetes_logs
+     config: {}
+     rawConfig: null
+   ...
++  customConfig:
++    ...
++    sources:
++      ...
++      kubernetes_logs:
++        type: kubernetes_logs
++    ...
+```
+
+### vectorSource
+
+The `extraContainerPorts` or `service` key should be used to expose the port configured in `customConfig`.
+
+```diff title="values.yaml"
+   vectorSource:
++    enabled: false
+     sourceId: vector
+     listenAddress: "0.0.0.0"
+     listenPort: "9000"
+     config: {}
+     nodePort: null
+     rawConfig: null
+   ...
++  customConfig:
++    ...
++    sources:
++      ...
++      vector:
++        type: vector
++        address: 0.0.0.0:9000
++    ...
+```
+
+### vectorSink
+
+```diff title="values.yaml"
+   vectorSink:
++    enabled: false
+     sinkId: vector_sink
+     inputs: ["kubernetes_logs"]
+     host: vector
+     port: "9000"
+     config: {}
+     rawConfig: null
+   ...
++  customConfig:
++    ...
++    sinks:
++      ...
++      vector_sink:
++        type: vector
++        inputs: ["kubernetes_logs"]
++        address: vector:9000
++    ...
+```
+
+### internalMetricsSource
+
+```diff title="values.yaml"
+   internalMetricsSource:
++    enabled: false
+     sourceId: internal_metrics
+     config: {}
+     rawConfig: null
+   ...
++  customConfig:
++    ...
++    sources:
++      ...
++      internal_metrics:
++        type: internal_metrics
++    ...
+```
+
+### hostMetricsSource
+
+Users should set the PROCFS_ROOT and SYSFS_ROOT environment variables, as well as mounting the
+required hostPaths.
```

Comment:
We could _try_ and detect `type: host_metrics` and include them dynamically still?